### PR TITLE
[WIP] Add buildings from POC

### PR DESCRIPTION
**Pull request by gjn**
This is an attempt to add buildings from the POC to our current application. I've duplicated code and buildings from the POC. Unfortunately, it doesn't show any buildings/objects yet. There is not error in the console whatsoever. Buildings files are loaded.

@gberaudo Do you have an idea why it doesn't work? Probably the transformation matrix is wrong. I have no idea in which coordinates the gltf files are, so there's probably a mismatch there.

### CODE REVIEW

File: src/js/GaCesium.js [hunk 1]
```diff
@@ -24,6 +24,109 @@ var GaCesium = function(map, gaPermalink, gaLayers, gaGlobalOptions, $q) {
...
+        var pointed = scene.globe.pick(ray, scene);
+        if (!Cesium.defined(pointed)) return;
+
+        var tile = [];
+        var bottom = Cesium.Ellipsoid.WGS84.cartesianToCartographic(pointed);
+        for (var i = 0; i < tiles.length; ++i) {
+          if (!Cesium.Rectangle.contains(tiles[i].rectangle, bottom)) continue;
+          tile.push(tiles[i]);
+        }
+        tile.forEach(function(tile) {
```
Comment (loicgasser) [inline src/js/GaCesium.js]: probably a little confusing to use tile forEach tile

File: src/js/GaCesium.js [hunk 2]
```diff
@@ -24,6 +24,109 @@ var GaCesium = function(map, gaPermalink, gaLayers, gaGlobalOptions, $q) {
...
+  // https://raw.githubusercontent.com/geoadmin/3d-testapp/198e5c3f5c5d22841d78183585874b4b5c85ddf4/poc.js?token=ACdAHY4w9qWFM6-g6ZISbNJWkhsLbEG7ks5WL2p0wA%3D%3D
+  //// This is a very simplified loading strategy for building tiles
+  var tileModelProvider = {
+     add: function(swCorner, neCorner, gltf) {
+        var rectangle = Cesium.Rectangle.fromCartographicArray([
+          Cesium.Cartographic.fromDegrees.apply(null, swCorner),
+          Cesium.Cartographic.fromDegrees.apply(null, neCorner)]
+        );
+        var center = Cesium.Rectangle.center(rectangle);
+        center.height = Math.max(swCorner[2], neCorner[2]) + 1400;
```
Comment (loicgasser) [inline src/js/GaCesium.js]: 1400? magic number

**Comment by gberaudo** [comment]
Indeed, there is an issue related to the transform matrix: the buliding tiles are upside down.
We add a similar issue once, which we fixed by converting again the collada to gltfs.
I will compile an updated converter and try out.

File: src/js/GaCesium.js (hunk 2 above)
Comment (loicgasser) [inline src/js/GaCesium.js]: by the way this is dead code.

**Comment by loicgasser** [comment]
buildings

Adding 25meters to the height does help, it seems that the initial extent are wrong,.

**Comment by gberaudo** [comment]
eastnorth25
`Corrected` with 25m: the buildings are upside down and the tile is not at the right place; a mirror transformation along the left axis would make the tile `fit`.

northdown0
Using a `Cesium.Transforms.northEastDownToFixedFrame` transform, the buildings are oriented up but the tile is not at the right place; a -90 degrees rotation around the red point would make the tile `fit`.

**Comment by loicgasser** [comment]
Lol it's a mess, is it a gLTF version issue? Why would the same files work in POC but not using the current geoadmin viewer? Sure we can do the transforms for demo purposes.

**Pull request closed without merging**